File: flash-runtime-engine/src/flash/system/Console.cpp
#include <flash/runtime/Console.h>
#include <flash/system/Console.h>

namespace flash::system::Console {
    void logMessage(const String &string) {
        runtime::Console::logMessage(string.source(), string.length());
    }

    void logInformation(const String &string) {
        runtime::Console::logInformation(string.source(), string.length());
    }

    void logWarning(const String &string) {
        runtime::Console::logWarning(string.source(), string.length());
    }

    void logError(const String &string) {
        runtime::Console::logError(string.source(), string.length());
    }

    void logStackTrace() {
        runtime::Console::logStackTrace();
    }
}

File: flash-runtime-engine/ref/Console.js
function FlashConsole() {
    let _runtime = null;

    this.logMessage = function(stringPtr, stringLen) {
        console.log(_runtime.readString(stringPtr, stringLen));
    };
    this.logInformation = function(stringPtr, stringLen) {
        console.info(_runtime.readString(stringPtr, stringLen));
    };
    this.logWarning = function(stringPtr, stringLen) {
        console.warn(_runtime.readString(stringPtr, stringLen));
    };
    this.logError = function(stringPtr, stringLen) {
        console.error(_runtime.readString(stringPtr, stringLen));
    };
    this.logStackTrace = function() {
        console.trace();
    };

    this.compose = function(runtime) {
        _runtime = runtime;
    };
    this.dispose = function() {
        _runtime = null;
    };

    return this;
}
